specs/lib/gltfPipelineSpec.js

```diff
@@ -0,0 +1,135 @@
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+    
+    it('optimizes a gltf JSON given a resourcePath', function() {
+        var gltf;
+        var gltfCopy;
+        var options = { 'resourcePath' : path.dirname(gltfPath) };
+        fs.readFile(gltfPath, function(err, data) {
+            if (err) {
+                throw err;
+            }
+            gltf = JSON.parse(data);
+            gltfCopy = clone(gltfCopy); 
+            addPipelineExtras(gltf);
+            
+            processJSON(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a gltf JSON with extras', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSONWithExtras(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+    
+    it('optimizes a binary glTF JSON with extras', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(glbPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSONWithExtras(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a glTF file', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = gltf;
+            processFile(gltfPath, options, function (gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('optimizes a glb file', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(glbPath, function(gltf) {
+            gltfCopy = gltf;
+            processFile(glbPath, options, function (gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+
+    it('will write a file to the correct directory', function(done) {
+        var spy = spyOn(fs, 'writeFile').and.callFake(function(file, data, callback) {
+            callback();
+        });
+        var options = {};
+        processFileToDisk(gltfPath, outputPath, options, function() {
+            expect(spy).toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it('will write a file', function(done) {
+        var spy = spyOn(fs, 'writeFile').and.callFake(function(file, data, callback) {
+            callback();
+        });
+        var options = {};
+        processFileToDisk(gltfPath, outputPath, options, function() {
+            expect(spy).toHaveBeenCalled();
+            done();
+        });
+    });
```

Comment:
I am having issues with merging these two tests into one where it is throwing: `Error: EISDIR: illegal operation on a directory, open 'output/output'
    at Error (native)`

I initially thought this was because it was trying to create a new directory in a path that already exists, but even after adding an option to not create a new directory I am still getting the same error.

Any thoughts? Should I just leave it as separate tests?